### PR TITLE
Add an endpoint for notifying us of clientside exceptions.

### PR DESCRIPTION
**Pull request by oponder**
This goes towards https://github.com/giantswarm/giantswarm/issues/6715.

We want to be notified when an error occurs in Happa (and perhaps gsctl too) and not having to wait until a client tells us about it.

### CODE REVIEW

File: spec/spec.yaml [hunk 1]
```diff
@@ -3071,3 +3071,38 @@ paths:
           description: error
           schema:
             $ref: "./definitions.yaml#/definitions/V4GenericResponse"
+
+  /v5/exception-notifications/:
+    post:
+      operationId: createExceptionNotification
+      x-internal: true
+      tags:
+        - exception notification
```
Comment (marians) [inline spec/spec.yaml]: This tag has to be defined in the `tags` section. And it should be plural.
```suggestion
        - exception notifications
```

File: spec/spec.yaml [hunk 2]
```diff
@@ -3071,3 +3071,38 @@ paths:
...
+      summary: Notifies Giant Swarm that an error has occured in one of our clients.
+      description: |
+        This endpoint is used to notify Giant Swarm that an error has occured in one of our clients (Like our Web UI or gsctl). It is not intended to be called manually, our clients are configured to report errors when they occur.
```
Comment (marians) [inline spec/spec.yaml]:
```suggestion
        This endpoint is used to notify Giant Swarm that an error has occured in one of our clients (like our Web UI or gsctl). It is not intended to be called manually, our clients are configured to report errors when they occur.
```

File: spec/spec.yaml [hunk 3]
```diff
@@ -3071,3 +3071,38 @@ paths:
...
+
+        Find us in your slack support channel if you want to contact us about any immediate issues.
```
Comment (marians) [inline spec/spec.yaml]:
```suggestion
        Find us in your Slack support channel if you want to contact us about any immediate issues.
```

File: spec/spec.yaml [hunk 4]
```diff
@@ -3071,3 +3071,38 @@ paths:
...
+        "200":
+          description: Exception notification created
+          schema:
+            $ref: "./definitions.yaml#/definitions/V4GenericResponse"
+          examples:
+            application/json:
+              {
+                "code": "RESOURCE_CREATED",
+                "message": "An exception notification has been sent to Giant Swarm."
+              }
```
Comment (marians) [inline spec/spec.yaml]: Please also add a `default` response description. This is requited to catch all other response codes, like 500.

**Comment by oponder** [comment]
re-ping @marians, fishing for approval from someone. expanding reviewrs to @giantswarm/sig-front-end  and @giantswarm/team-batman

File: spec/spec.yaml [hunk 5]
```diff
@@ -3071,3 +3074,42 @@ paths:
           description: error
           schema:
             $ref: "./definitions.yaml#/definitions/V4GenericResponse"
+
+  /v5/exception-notifications/:
+    post:
+      operationId: createExceptionNotification
```
Comment (marians) [inline spec/spec.yaml]: just noticed: In the operationId we use the verb `add` in cases like `addCluster`. Maybe align this here? (Might be that we are using `create` in some cases, too)

Comment (oponder) [reply spec/spec.yaml]: yeah we have `create` and `add` used without any specific rule. I'll change this to `add` here though while we can.

**Pull request merged**